ssz/src/main/java/org/ethereum/beacon/ssz/access/AccessorResolverRegistry.java

```diff
@@ -0,0 +1,136 @@
+package org.ethereum.beacon.ssz.access;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.ethereum.beacon.ssz.SSZSchemeException;
+import org.ethereum.beacon.ssz.access.basic.SubclassCodec;
+import org.ethereum.beacon.ssz.access.list.SubclassListAccessor;
+import org.ethereum.beacon.ssz.annotation.SSZSerializable;
+
+public class AccessorResolverRegistry implements AccessorResolver {
+
+  private Map<Class, List<CodecEntry>> registeredClassHandlers = new HashMap<>();
+  List<SSZListAccessor> listAccessors;
+  List<SSZContainerAccessor> containerAccessors;
+
+  public AccessorResolverRegistry withContainerAccessors(List<SSZContainerAccessor> containerAccessors) {
+    this.containerAccessors = containerAccessors;
+    return this;
+  }
+
+  public AccessorResolverRegistry withListAccessors(List<SSZListAccessor> listAccessors) {
+    this.listAccessors = listAccessors;
+    return this;
+  }
+
+  public AccessorResolverRegistry withBasicCodecs(List<SSZCodec> codecs) {
+    for (SSZCodec codec : codecs) {
+      registerCodec(codec);
+    }
+    return this;
+  }
+
+  @Override
+  public Optional<SSZListAccessor> resolveListAccessor(SSZField field) {
+    if (!SubclassListAccessor.getSerializableClass(field.getRawClass()).equals(field.getRawClass())) {
+      return resolveListAccessor(SubclassListAccessor.getSerializableField(field))
+          .map(SubclassListAccessor::new);
+    }
+
+    Object accessor = getAccessorFromAnnotation(field);
+    if (accessor instanceof SSZListAccessor) {
+      return Optional.of((SSZListAccessor) accessor);
+    }
+    return listAccessors.stream().filter(a -> a.isSupported(field)).findFirst();
+  }
+
+  @Override
+  public Optional<SSZContainerAccessor> resolveContainerAccessor(SSZField field) {
+    Object accessor = getAccessorFromAnnotation(field);
+    if (accessor instanceof SSZContainerAccessor) {
+      return Optional.of((SSZContainerAccessor) accessor);
+    }
+
+    return containerAccessors.stream().filter(a -> a.isSupported(field)).findFirst();
+  }
+
+  private Object getAccessorFromAnnotation(SSZField field) {
+    SSZSerializable annotation = field.getRawClass().getAnnotation(SSZSerializable.class);
+    if (annotation == null || annotation.accessor() == void.class) {
+      return null;
+    }
+
+    try {
+      return annotation.accessor().newInstance();
+    } catch (InstantiationException | IllegalAccessException e) {
+      throw new SSZSchemeException("Couldn't create codec instance " + annotation.accessor() + " for field " + field);
+    }
+  }
+
+  @Override
+  public SSZCodec resolveBasicTypeCodec(SSZField field) {
+    Object accessor = getAccessorFromAnnotation(field);
```

Comment:
maybe use `Optional ` in `getAccessorFromAnnotation` return type instead? Looks a bit dirty.

Reply:
Refactored this place